module/VuFind/src/VuFind/View/Helper/Root/AbstractClassBasedTemplateRenderer.php

```diff
@@ -93,14 +102,22 @@ protected function resolveClassTemplate($template, $className,
     protected function renderClassTemplate($template, $className, $context = [])
     {
         // Set up the needed context in the view:
-        $contextHelper = $this->getView()->plugin('context');
-        $oldContext = $contextHelper($this->getView())->apply($context);
+        $view = $this->getView();
+        $contextHelper = $view->plugin('context');
+        $oldContext = $contextHelper($view)->apply($context);
 
-        // Render the template for the current class:
-        $html = $this->resolveClassTemplate($template, $className);
+        // Find the template for the current class:
+        if (!isset($this->templateCache[$className][$template])) {
```

Comment:
I feel like the cache functionality might be cleaner in its own method -- i.e. `$this->getCachedClassTemplate()`; I think it would improve readability of this method if the caching details were separated.

Reply:
Done.